src/Altinn.App.Core/Infrastructure/Clients/Email/EmailNotificationClient.cs

```diff
@@ -0,0 +1,125 @@
+        .CreateCounter("altinn_app_notification_order_request_count", "Number of notification order requests.", labelNames: ["type", "result"]);
+
+    private readonly JsonSerializerOptions _jsonSerializerOptions = new()
+    {
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase
+    };
+
+    /// <inheritdoc/>
+    /// <exception cref="EmailNotificationException"/>
+    public async Task<EmailOrderResponse> Order(EmailNotification emailNotification, CancellationToken ct)
+    {
+        var startTime = DateTime.UtcNow;
+        var timer = Stopwatch.StartNew();
+
+        using var httpClient = _httpClientFactory.CreateClient();
+
+        HttpResponseMessage? httpResponseMessage = null;
+        string? httpContent = null;
+        EmailOrderResponse? orderResponse = null;
+        try
+        {
+            var application = await _appMetadata.GetApplicationMetadata();
+
+            var uri = _platformSettings.NotificationEndpoint.TrimEnd('/') + "/api/v1/orders/email";
+            var body = JsonSerializer.Serialize(emailNotification, _jsonSerializerOptions);
+
+            using var httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, uri)
+            {
+                Content = new StringContent(body, new MediaTypeHeaderValue("application/json")),
+            };
+            httpRequestMessage.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            httpRequestMessage.Headers.Add(
+                "PlatformAccessToken",
+                _accessTokenGenerator.GenerateAccessToken(application.Org, application.AppIdentifier.App)
+            );
+
+            httpResponseMessage = await httpClient.SendAsync(httpRequestMessage, ct);
+            httpContent = await httpResponseMessage.Content.ReadAsStringAsync(ct);
+            if (httpResponseMessage.IsSuccessStatusCode)
+            {
+                orderResponse = JsonSerializer.Deserialize<EmailOrderResponse>(httpContent);
+                if (orderResponse is null)
+                    throw new InvalidOperationException("Couldn't deserialize email notification order response.");
+
+                _orderCount.WithLabels("email", "success").Inc();
+            }
+            else
+            {
+                throw new HttpRequestException("Got error status code for email notification order");
+            }
+            return orderResponse;
+        }
+        catch (Exception e)
+        {
+            _orderCount.WithLabels("email", "error").Inc();
+            var ex = new EmailNotificationException("Something went wrong when processing the email order, see inner exception for details.", e);
+            ex.Data.Add("responseContent", httpContent);
+            ex.Data.Add("responseStatusCode", httpResponseMessage?.StatusCode.ToString());
+            ex.Data.Add("responseReasonPhrase", httpResponseMessage?.ReasonPhrase);
+            _telemetryClient.TrackException(ex);
```

Comment:
Is this manual work with `TelemetryClient` really required? In other parts of the code we just inject an `ILogger<ClassName>` and do `_logger.LogError(ex, "Comment about were the exception happened")`, and it seems to be picked up by application insights just fine. For dependencies we rely on the automatic instrumentation of httpClient, that also (to my knowledge) does most of what you seem to attempt (apart from hiding the actual server name with a more pretty `Altinn.Notifications.OrderEmailNotification`). Won't the current implementation show up as two dependencies that are really the same?

Reply:
TrackException not necessary as this i done by default, tested to verify. TrackDependency is nice as the class does not only call an http endpoint, it gets applicationdata and fetches a plattform token as well.